module3/task3.py

```diff
@@ -0,0 +1,250 @@
+
+    def notify_im_leaving(self, employee):
+        """ En employee should call this method when leaving a company """
+        if employee in self.employees:
+            self.employees.remove(employee)
+
+    def do_tasks(self, employee):
+        """
+        Engineer should call this method when he is working.
+        Company should withdraw 10 money from a personal account and return
+        them to engineer. That will be a payment
+        :rtype: int
+        """
+        # make sure engineer is employed to this company
+        # check employee is Engineer
+        salary_for_engineer = 10
+        if employee in self.employees and isinstance(employee, Engineer):
+            self.__money -= salary_for_engineer
+            print(f"Employee {employee.name} has done task and got {salary_for_engineer}")
+            return salary_for_engineer
+
+    def write_reports(self, employee):
+        """
+        Manager should call this method when he is working.
+        Company should withdraw 12 money from a personal account and return
+        them to manager. That will be a payment
+        :rtype: int
+        """
+        # make sure manager is employed to this company
+        # check employee is Manager
+        salary_for_manager = 12
+        if employee in self.employees and isinstance(employee, Manager):
+            self.__money -= salary_for_manager
+            print(f"Employee {employee.name} has done task and got {salary_for_manager}")
+            return salary_for_manager
+
+    def make_a_party(self):
+        """ Party time! All employees get 5 money """
+        # make sure a company is not a bankrupt before and after the party
+        # call employee.bonus_to_salary()
+        if not self.is_bankrupt and len(self.employees) * 5 < self.__money:
+            print(f"{self.name} is organizing a party!")
+            for employee in self.employees:
+                employee.bonus_to_salary(self)
+        else:
+            print("It is not possible to organize a party now, because of not enough money in the budget")
+
+    def show_money(self):
+        """ Displays amount of money that company has """
+        print(f"Current money for {self.name} is {self.__money}")
+
+    def go_bankrupt(self):
+        """
+        Declare bankruptcy. Company money are drop to 0.
+        All employees become unemployed.
+        """
+        self.__money = 0
+        print(f"{self.name} is likely to be a bankrupt")
+        for employee in self.employees:
+            self.dismiss_employee(employee)
+
+
+    @property
+    def is_bankrupt(self):
+        """ returns True or False """
+        return self.__money <= 0
+
+
+    def __repr__(self):
+        return 'Company (%s)' % self.name
+
+
+class Person(object):
+    """ Represents any person """
+
+    def __init__(self, name, age, sex=None, address=None):
+        self.name = name
+        self.age = age
+        self.sex = sex if sex is not None else '<not specified>'
```

Comment:
Just FYI, you could also use "or" operator

= sex or '<not specified>'

but please pay attention that in case of sex = False for example, it'd work differently compared to yours implementation

Reply:
Same as above